packages/osd-test/src/legacy_opensearch/opensearch_test_config.js

```diff
@@ -56,30 +55,39 @@ export const opensearchTestConfig = new (class OpenSearchTestConfig {
   getUrlParts() {
     // Allow setting one complete TEST_OPENSEARCH_URL for opensearch like https://opensearch:changeme@example.com:9200
     if (process.env.TEST_OPENSEARCH_URL) {
-      const testOpenSearchUrl = url.parse(process.env.TEST_OPENSEARCH_URL);
+      const testOpenSearchUrl = new URL('', process.env.TEST_OPENSEARCH_URL);
```

Comment:
Should this actually be 
```suggestion
      const testOpenSearchUrl: URL = new URL('', process.env.TEST_OPENSEARCH_URL);
```
?

I'd expect the typing to come through automatically, but I think TS is viewing `testOpenSearchUrl` as `any`, which is why you find yourself needing the `as string` casts. If it's a `URL` type, then TS will know that the `toString()` method only generates strings, and you can get rid of all those.

Reply:
Are you referring [this `as string`](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2910/files#diff-2907cf12204194416692b729ff7c3f39e7abdce7ab1bcfe012335f247a3995a8R85)? I try to add `: URL`, but it doesn't work in my local. I think TS know `testOpenSearchUrl` is an `URL` type, because we construct a new object here. The root cause of add `as string` is [the defaultValue](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2910/files/a846ba14238a071812b5083e47fc2e8cd54574f6#diff-e778dd71aaf98aafc6d39c0cb93648ab1bf357a98296c726a949eed62064cf4aR120) is any, the `loadash.get` will return any type too. How about we change the as string to following code:
```tsx
        const serverUrl = config.get('servers.opensearchDashboards.serverUrl');
        if (typeof serverUrl === 'string') {
          opensearchDashboardsUrl = serverUrl;
        }
```